lzy/worker/src/main/java/ai/lzy/worker/StreamQueue.java

```diff
@@ -1,67 +1,163 @@
 package ai.lzy.worker;
 
+import ai.lzy.v1.common.LMO;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.FormattedMessage;
+import org.apache.logging.log4j.util.Strings;
 
 import java.io.*;
-import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+
+import static java.util.Arrays.copyOfRange;
 
 public class StreamQueue extends Thread {
-    private final OutputStream out;
-    private final LinkedBlockingQueue<InputStream> inputs = new LinkedBlockingQueue<>();
+    private final ArrayList<OutputStream> outs = new ArrayList<>();
+    private final LinkedBlockingQueue<Input> inputs = new LinkedBlockingQueue<>();
     private final Logger logger;
     private final String streamName;
     private final AtomicBoolean stopping = new AtomicBoolean(false);
+    @Nullable private final KafkaProducer<String, byte[]> kafkaClient;
+    @Nullable private final String topic;
 
-    public StreamQueue(OutputStream out, Logger log, String streamName) {
-        this.out = out;
+    public StreamQueue(@Nullable LMO.KafkaTopicDescription topic, Logger log, String streamName) {
         this.logger = log;
         this.streamName = streamName;
+        if (topic != null) {
+            var props = new Properties();
+            props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
```

Comment:
let's introduce some KafkaHelper with `configure` methods
and move all these literal strings to it